deeppavlov/models/nemo/common.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+from io import BytesIO
+from logging import getLogger
+from pathlib import Path
+from typing import Union
+
+import nemo
+import torch
+from nemo.backends.pytorch import DataLayerNM
+from torch.utils.data import Dataset, DataLoader
+
+from deeppavlov.core.common.file import read_yaml
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.models.component import Component
+from deeppavlov.core.models.serializable import Serializable
+from deeppavlov.core.commands.utils import expand_path
+
+log = getLogger(__name__)
+
+
+@register('ascii_to_bytesIO')
+def ascii_to_bytes_io(batch: Union[str, list]) -> Union[BytesIO, list]:
+    """Recursively searches for strings in the input batch and converts them into the base64-encoded bytes wrapped in
+    Binary I/O objects.
+
+    Args:
+        batch: A string or an iterable container with strings at some level of nesting.
+
+    Returns:
+        The same structure where all strings are converted into the base64-encoded bytes wrapped in Binary I/O object.
+
+    """
+    if isinstance(batch, str):
+        return BytesIO(base64.decodebytes(batch.encode()))
+
+    return list(map(ascii_to_bytes_io, batch))
+
+
+@register('bytesIO_to_ascii')
```

Comment:
I think we should mention `base64` in the name somewhere. Maybe `base64_encode_bitesIO`?